### PR TITLE
Bug/duplicate queries

### PR DESCRIPTION
**Pull request by intoeetive**
avoid some duplicate queries in CP 

use member model from session where appropriate
EECORE-1641

turn the list of model properties into static variable 
EECORE-1641
 
fix cache check for compatible fieldtypes

save assigned modules to session cache

### CODE REVIEW

File: system/ee/ExpressionEngine/Addons/textarea/ft.textarea.php [hunk 1]
```diff
@@ -31,18 +31,17 @@ public function validate($data)
...
-                ->first();
+            && ! ee()->session->cache(__CLASS__, 'markitup_initialized')
+        ) {
 
             // channel form only uses formatting buttons in the {custom_fields}{/custom_fields}
             // tag pair which does NOT call this method. This method is still called with {field:my_textarea}, though,
             // so we do need to at least avoid a fatal error if that tag exists and the form allows guest authors.
-            if ($member) {
-                $buttons = $member->getHTMLButtonsForSite(ee()->config->item('site_id'));
+            if (ee()->session->userdata('member_id') != 0) {
```
Comment (matthewjohns0n) [inline system/ee/ExpressionEngine/Addons/textarea/ft.textarea.php:43]: Is this always 0? Or can it be `NULL` if not logged in?

File: system/ee/ExpressionEngine/Model/Content/VariableColumnGateway.php [hunk 2]
```diff
@@ -32,7 +32,11 @@ public function getFieldList($cached = true)
 
         $known = parent::getFieldList();
 
-        return $this->_field_list_cache = array_merge($known, $all);
+        if (!is_array(static::$_field_list_cache)) {
+            static::$_field_list_cache = [];
+        }
+
+        return static::$_field_list_cache[get_class($this)] = array_merge($known, $all);
```
Comment (matthewjohns0n) [inline system/ee/ExpressionEngine/Model/Content/VariableColumnGateway.php:39]: Just wondering, was the changed from `$this->_field_list_cache` to `static::$_field_list_cache[get_class($this)]` to support caching multiple models?

**Review by bryannielsen — APPROVED** [review]

File: system/ee/ExpressionEngine/Addons/textarea/ft.textarea.php (hunk 1 above)
Comment (intoeetive) [reply system/ee/ExpressionEngine/Addons/textarea/ft.textarea.php:43]: I believe it's always casted to int

File: system/ee/ExpressionEngine/Model/Content/VariableColumnGateway.php (hunk 2 above)
Comment (intoeetive) [reply system/ee/ExpressionEngine/Model/Content/VariableColumnGateway.php:39]: At this point, I don't quite remember =)
But it looks so, based on the code.
The main reason to change was the `static::` part

Comment (matthewjohns0n) [reply system/ee/ExpressionEngine/Model/Content/VariableColumnGateway.php:39]: Okay, makes sense 👍

**Review by matthewjohns0n — APPROVED** [review]

**Pull request merged**